code/src/java/pcgen/cdom/reference/PatternMatchingReference.java

```diff
@@ -260,4 +256,17 @@ public String getChoice()
 	{
 		return null;
 	}
+
+	@Override
+	public Class<T> getReferenceClass()
+	{
+		return all.getReferenceClass();
+	}
+
+	@Override
+	public String getReferenceDescription()
+	{
+		//TODO Is this really what I want?!?
+		return all.getReferenceDescription();
```

Comment:
It's a good question :-).  I'm actually wondering about the DRY pattern being broken, e.g. the `getReferenceClass` and `getReferenceDescription` methods both pass through to an underlying object for this class and all others like it.  I wonder if it could be hauled up somehow...

Reply:
This was my way of saying "This isn't done, hopefully someone catches this".. so thanks for catching it... all the items will have polished descriptions now.

As far as the underlying objects, there is a pretty major philosophical debate here that I think deserves an off-line discussion.  There are reasons the tools that catch these delegation behaviors refer to the checking rules as "controversial"... The underlying information and timing of object construction requires one or more good design patterns to be violated, the question is which one.